src/wallet/wallet.cpp

```diff
@@ -3670,6 +3715,27 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
 
         walletInstance->SetBestChain(chainActive.GetLocator());
     }
+    else if (IsArgSet("-disablehot")) {
+        bool disableHot = GetBoolArg("-disablehot", DEFAULT_DISABLE_HOT_WALLET);
+        LOCK(walletInstance->cs_wallet);
+        if (walletInstance->vchDefaultKey.IsValid() && disableHot) {
```

Comment:
I think you should use `fFirstRun` here instead of `walletInstance->vchDefaultKey.IsValid()`

Reply:
This should at least be abstracted somehow. vchDefaultKey's existence is not exactly identical to "does this wallet have keys?"